google/auth/aio/transport/__init__.py

```diff
@@ -79,3 +79,53 @@ async def read(self) -> bytes:
     async def close(self):
         """Close the response after it is fully consumed to resource."""
         raise NotImplementedError("close must be implemented.")
+
+
+class Request(metaclass=abc.ABCMeta):
+    """Interface for a callable that makes HTTP requests.
+
+    Specific transport implementations should provide an implementation of
+    this that adapts their specific request / response API.
+
+    .. automethod:: __call__
+    """
+
+    @abc.abstractmethod
+    async def __call__(
+        self,
+        url: str,
+        method: Optional[str],
+        body: bytes,
+        headers: Mapping[str, str],
```

Comment:
@clundin25 
Do we want to change the `Response.headers` type to a `Mapping` instead of a `Dict` to match this?

Reply:
If required, will be addressed in the cleanup PR towards the end.